hexagonal/src/test/java/com/iluwatar/hexagonal/domain/LotteryTicketTest.java

```diff
@@ -49,4 +56,77 @@ void testEquals() {
     LotteryTicket ticket3 = new LotteryTicket(new LotteryTicketId(), details3, numbers3);
     assertNotEquals(ticket1, ticket3);
   }
+
+  @Test
+  void sameObject() {
+    PlayerDetails details = new PlayerDetails("bob@foo.bar", "1212-121212", "+34332322");
+    LotteryNumbers numbers = LotteryNumbers.create(new HashSet<>(Arrays.asList(1, 2, 3, 4)));
+    LotteryTicket ticket = new LotteryTicket(new LotteryTicketId(), details, numbers);
+
+    assertEquals(ticket, ticket);
+  }
+
+  @Test
+  void otherNull() {
+    PlayerDetails details = new PlayerDetails("bob@foo.bar", "1212-121212", "+34332322");
+    LotteryNumbers numbers = LotteryNumbers.create(new HashSet<>(Arrays.asList(1, 2, 3, 4)));
+    LotteryTicket ticket = new LotteryTicket(new LotteryTicketId(), details, numbers);
+
+    assertNotEquals(ticket, null);
```

Comment:
Could it be clearer to use assertFalse(ticket.equals(null))? The same goes for the rest of the assertions.

Reply:
Or in this particular case: assertNotNull!

Reply:
We need to use some assertion that tests `LotteryTicket::equals`. Therefore I believe `assertNotNull` could be problematic.

I've now changed to use `assertFalse`/`assertTrue` though since it makes it more explicit what is tested. It might result in worse test failure messages though.